gio/src/test_util.rs

```diff
@@ -23,3 +23,22 @@ pub fn run_async<T: Send + 'static, Q: FnOnce(Sender<T>, MainLoop) + Send + 'sta
 
     rx.recv().unwrap()
 }
+
+#[allow(dead_code)]
+pub fn run_async_local<T: 'static, Q: FnOnce(Sender<T>, MainLoop) + Send + 'static>(start: Q) -> T {
```

Comment:
What's with this? :)

Reply:
I did that because using `invoke` i would need `Value` to implement `Send`, but it does not since the raw pointer to `GValue` is not `Send`:
```
error[E0277]: `*mut c_void` cannot be sent between threads safely
   --> gio/src/task.rs:228:15
    |
228 |         match run_async(|tx, l| {
    |               ^^^^^^^^^ `*mut c_void` cannot be sent between threads safely
    | 
   ::: gio/src/test_util.rs:7:21
    |
7   | pub fn run_async<T: Send + 'static, Q: FnOnce(Sender<T>, MainLoop) + Send + 'static>(
    |                     ---- required by this bound in `run_async`
    |
    = help: within `std::result::Result<Value, glib::Error>`, the trait `Send` is not implemented for `*mut c_void`
    = note: required because it appears within the type `GValue_data`
    = note: required because it appears within the type `[GValue_data; 2]`
    = note: required because it appears within the type `GValue`
    = note: required because it appears within the type `Value`
    = note: required because it appears within the type `std::result::Result<Value, glib::Error>`

```

Reply:
Makes sense